docs/cookbook/custom-promotion-rule.rst

```diff
@@ -0,0 +1,96 @@
+How to add a custom promotion rule?
+===================================
+
+Adding new, custom rules to your shop is a common usecase. You can imagine for instance, that you have some customers
+in your shop that you distinguish as premium. And for these premium customers you would like to give special promotions.
+For that you will need a new PromotionRule that will check if the customer is premium or not.
+
+Create a new promotion rule
+---------------------------
+
+The new Rule needs a RuleChecker class:
+
+.. code-block:: php
+
+    <?php
+
+    namespace AppBundle\Promotion\Checker\Rule;
+
+    use Sylius\Component\Promotion\Checker\Rule\RuleCheckerInterface;
+    use Sylius\Component\Promotion\Model\PromotionSubjectInterface;
+
+
+    class PremiumCustomerRuleChecker implements RuleCheckerInterface
+    {
+        const TYPE = 'premium_customer';
```

Comment:
Each rule checker has this constant to be able to create a new rule with type programmatically :)
```
$rule->setType(PremiumCustomerRuleChecker::TYPE);
```